src/main/java/com/gluonhq/substrate/util/Strings.java

```diff
@@ -0,0 +1,51 @@
+package com.gluonhq.substrate.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class Strings {
+
+    /**
+     * Replaces keys within the template with values using context function
+     * Throws IllegalArgumentException if the key found in the template has no corresponding value
+     * @param template template to use
+     * @param context context to get values from
+     * @return resulting string with keys replaced with corresponding values
+     */
+    public static String substitute(String template, Function<String,String> context) {
+        Objects.requireNonNull(context, "Context is required");
+        Pattern pattern = Pattern.compile("\\$\\{(.+?)}");
```

Comment:
shouldn't be this pattern a static field?

Reply:
True - done